### ISSUE TITLE
Uploading file that changes units does not seem to invalidate cache so subsequent downloads are broken

### ISSUE THREAD

**Issue by dwaynebailey** [translate/pootle#3746]
1. Download a file
2. Make changes to the file
3. Upload the file
4. Inspect and units are correctly changed
5. Download file again - we have the old file without any of the updates that we got via upload

**Comment by jleclanche** [comment]
I've been looking at this all evening, I cannot figure out why the store isn't getting updated. It's a problem in the Pootle/TTK binding. @julen @unho feedback appreciated.

**Comment by dwaynebailey** [comment]
My gut feel is that its nothing to do with any 'binding'.  I'm simply getting old data that does not include recently submitted or recently uploaded values.  So my gut feeling is that somehow when we export we're not requesting a brand new serialization but are getting a cached version.

**Comment by julen** [comment]
In `Store` instances, `file` references the on-disk file, and its `store` property holds a reference to an on-disk parsed file.

In the new export code, `serialize()` always returns [the string representation of the on-disk file](https://github.com/translate/pootle/blob/ac782c6ed3eda1c9fba1acbbe1b87c562c461b6f/pootle/apps/pootle_store/models.py#L1801), so the behavior described in the OP aligns with what I read in the code.

In other words, `serialize()` does zero work. I believe `serialize()` has good intentions, but it ends up trying to reinvent the wheel, as [the serialization step is already implemented in `convert()`](https://github.com/translate/pootle/blob/ac782c6ed3eda1c9fba1acbbe1b87c562c461b6f/pootle/apps/pootle_store/models.py#L1939-L1950).

I highly advise to write some test cases besides fixing the issue.

**Comment by jleclanche** [comment]
That convert code is just nasty... I'm not even sure how to clean this up properly.

**Comment by jordis** [comment]
Note that this issue exists even if the strings are changed/translated inside the Pootle IU, that is, string changes do not need to happen offline and no file download/upload is involved.
See: https://groups.google.com/d/topic/mozilla.dev.l10n/QUoBfpfr4xU/discussion
